cpp/src/Ice/CMakeLists.txt

```diff
@@ -0,0 +1,595 @@
+		NetworkProxy.cpp
+		NetworkProxy.h
+		NetworkProxyF.h
+		Object.cpp
+		ObjectAdapterFactory.cpp
+		ObjectAdapterFactory.h
+		ObjectAdapterFactoryF.h
+		ObjectAdapterI.cpp
+		ObjectAdapterI.h
+		ObserverHelper.cpp
+		OpaqueEndpointI.cpp
+		OpaqueEndpointI.h
+		OutgoingAsync.cpp
+		OutputStream.cpp
+		PluginManagerI.cpp
+		PluginManagerI.h
+		PropertiesAdminI.cpp
+		PropertiesAdminI.h
+		PropertiesI.cpp
+		PropertiesI.h
+		PropertyNames.cpp
+		PropertyNames.h
+		Protocol.cpp
+		ProtocolInstance.cpp
+		ProtocolInstance.h
+		ProtocolInstanceF.h
+		ProtocolPluginFacade.cpp
+		ProtocolPluginFacade.h
+		ProtocolPluginFacadeF.h
+		Proxy.cpp
+		ProxyFactory.cpp
+		ProxyFactory.h
+		Reference.cpp
+		Reference.h
+		ReferenceFactory.cpp
+		ReferenceFactory.h
+		ReferenceFactoryF.h
+		RegisterPluginsInit.cpp
+		RegisterPluginsInit.h
+		ReplyStatus.h
+		RequestHandler.cpp
+		RequestHandler.h
+		RequestHandlerFactory.cpp
+		RequestHandlerFactory.h
+		ResponseHandler.cpp
+		ResponseHandler.h
+		RetryQueue.cpp
+		RetryQueue.h
+		RetryQueueF.h
+		RouterInfo.cpp
+		RouterInfo.h
+		RouterInfoF.h
+		Selector.cpp
+		Selector.h
+		ServantManager.cpp
+		ServantManager.h
+		Service.cpp
+		SHA1.cpp
+		SharedContext.h
+		SliceChecksums.cpp
+		SlicedData.cpp
+		StreamSocket.cpp
+		StreamSocket.h
+		StringConverterPlugin.cpp
+		StringUtil.h
+		SysLoggerI.cpp
+		SysLoggerI.h
+		TcpAcceptor.cpp
+		TcpAcceptor.h
+		TcpConnector.cpp
+		TcpConnector.h
+		TcpEndpointI.cpp
+		TcpEndpointI.h
+		TcpTransceiver.cpp
+		TcpTransceiver.h
+		Thread.cpp
+		ThreadPool.cpp
+		ThreadPool.h
+		Timer.cpp
+		TraceLevels.cpp
+		TraceLevels.h
+		TraceLevelsF.h
+		TraceUtil.cpp
+		TraceUtil.h
+		Transceiver.cpp
+		Transceiver.h
+		TransceiverF.h
+		UdpConnector.cpp
+		UdpConnector.h
+		UdpEndpointI.cpp
+		UdpEndpointI.h
+		UdpTransceiver.cpp
+		UdpTransceiver.h
+		Value.cpp
+		ValueFactoryManagerI.cpp
+		ValueFactoryManagerI.h
+		VirtualShared.h
+		WSAcceptor.cpp
+		WSAcceptor.h
+		WSConnector.cpp
+		WSConnector.h
+		WSEndpoint.cpp
+		WSEndpoint.h
+		WSTransceiver.cpp
+		WSTransceiver.h
+	)
+endif()
+
+list(APPEND ICE_SLICE_DIRS "-I${SLICE_DIR}")
+list(APPEND ICE_SLICE_PARAMS ${COMPILE_SLICES_CPP_PARAMETERS} "Ice")
+
+foreach(slice ${ICE_SLICES})
+	get_filename_component(slice_name ${slice} NAME_WE)
+	set(output_files 
+		${GENERATED_SLICES_DIR}/Ice/${slice_name}.cpp 
+		${GENERATED_SLICES_DIR}/Ice/${slice_name}.h
+	)	
+	#function(CompileSlice slice2bin_params slice_include_paths ice_file output_dir output_file)
+	CompileSlice("${ICE_SLICE_PARAMS}"
+		"${ICE_SLICE_DIRS}"
+		${slice}
+		${GENERATED_SLICES_DIR}/Ice
+		"${output_files}"
+	)
+endforeach()
+
+install(DIRECTORY ${GENERATED_SLICES_DIR}/Ice 
+	DESTINATION ${GENERATED_H_INSTALL_DIR}
+	FILES_MATCHING PATTERN "*.h"
+)
+
+if(BUILD_ICE_CPP98)
+	set(ICE_TARGET ice)
+
+	add_library(${ICE_TARGET})
+
+	set_target_properties(${ICE_TARGET} 
+		PROPERTIES
+			OUTPUT_NAME Ice${zeroc-ice_VERSION_MAJOR}${zeroc-ice_VERSION_MINOR}
+			CXX_STANDARD 98
```

Comment:
This does not match the current build system where the C++98 mapping is built with the compiler's default.

Reply:
I'm not sure what you mean. MSVC should ignore this setting, but linux/mac might use it. This setting was based on the build documentation.

Reply:
We document that the C++98 mapping is built by default with the default -std, not -std=c++98. See https://doc.zeroc.com/ice/3.7/release-notes/known-issues-and-platform-notes#id-.KnownIssuesandPlatformNotesv3.7-C++98Mappingand-stdoptiononLinuxARM

Did you find other documentation that says something else?

Reply:
It might have been that I based that on folder structure in the VS solution/project sets. I can remove those settings and push later this evening.

Reply:
This has been addressed.